tachyon/crypto/sumcheck/multilinear/sumcheck_verifier.h

```diff
@@ -181,38 +181,54 @@ F InterpolateUniPoly(const std::vector<F>& poly, const F& evaluation_point) {
 #else
   size_t thread_nums = 1;
 #endif
-  thread_nums =
-      ((thread_nums * kParallelFactor) <= poly_size) ? thread_nums : 1;
 
-  size_t chunk_size = (poly_size + thread_nums - 1) / thread_nums;
-  size_t num_chunks = (poly_size + chunk_size - 1) / chunk_size;
+  struct Result {
+    F product;
+    F denom_up;
+    std::vector<F> evals;
+  };
 
-  std::vector<F> products(num_chunks, F::One());
-  std::vector<F> denom_ups(num_chunks, F::One());
-  std::vector<std::vector<F>> list_of_evals(num_chunks);
-  OMP_PARALLEL_FOR(size_t i = 0; i < num_chunks; ++i) {
-    size_t begin = i * chunk_size;
-    size_t len = (i == num_chunks - 1) ? poly_size - begin : chunk_size;
-    list_of_evals[i].reserve(len);
-    F check = F(begin);
-    for (size_t j = begin; j < begin + len; ++j) {
-      const F difference = evaluation_point - check;
-      list_of_evals[i].push_back(difference);
-      products[i] *= difference;
-      if (j > 1) {
-        denom_ups[i] *= check;
-      }
-      check += F::One();
+  F product;
+  F denom_up;
+  std::vector<F> evals;
+  {
+    std::vector<Result> results = base::ParallelizeMap(
+        poly_size,
+        [&evaluation_point](size_t len, size_t chunk_offset,
+                            size_t chunk_size) {
+          size_t begin = chunk_offset * chunk_size;
+          Result result;
+          result.product = F::One();
+          result.denom_up = F::One();
+          result.evals.reserve(len);
+          F check = F(begin);
+          for (size_t i = begin; i < begin + len; ++i) {
+            const F difference = evaluation_point - check;
+            result.evals.push_back(difference);
+            result.product *= difference;
+            if (i > 1) {
```

Comment:
`if (i > 1) {` is correct since `check` = `begin` as a field. In the case of `begin` = 0, `result.denom_up *= 0` would trash the value to stay at 0 all the time and in the case of `begin = 1`, `*= 1` is redundant.